docs/development/plugin-creation.md

```diff
@@ -9,6 +11,23 @@ If you create your own plugin:
 5. Subscribe to the [dokku development blog](http://progrium.com) to be notified about API changes and releases
 
 
+### Compilable Plugins (i.e. golang, java(?), c, etc.)
+The plugin developer is required to implement the `install` trigger such that it will output your built executable(s) in the correct directory structure to implement the plugin's desired command and/or trigger API. See [smoke-test-plugin](https://github.com/dokku/smoke-test-plugin) for an example.
```

Comment:
Best method for what? I linked to the smoke-test-plugin as an example instead of cluttering up the docs with code examples.